launcher/src/test/java/org/apache/brooklyn/launcher/CleanOrphanedLocationsLiveTest.java

```diff
@@ -0,0 +1,257 @@
+        PersistenceObjectStore objectStore = BrooklynPersistenceUtils.newPersistenceObjectStore(managementContext, persistenceLocation, persistenceDir,
+                PersistMode.AUTO, HighAvailabilityMode.HOT_STANDBY);
+
+        BrooklynMementoPersisterToObjectStore persister = new BrooklynMementoPersisterToObjectStore(
+                objectStore,
+                ((ManagementContextInternal)managementContext).getBrooklynProperties(),
+                managementContext.getCatalogClassLoader());
+
+        RebindManager rebindManager = managementContext.getRebindManager();
+
+        PersistenceExceptionHandler persistenceExceptionHandler = PersistenceExceptionHandlerImpl.builder().build();
+        ((RebindManagerImpl) rebindManager).setPeriodicPersistPeriod(Duration.ONE_SECOND);
+        rebindManager.setPersister(persister, persistenceExceptionHandler);
+    }
+
+    @Test
+    public void testSelectionWithOrphanedLocationsInData() {
+
+        Set<String> locationsToKeep = MutableSet.of(
+                "tjdywoxbji",
+                "pudtixbw89"
+        );
+
+        Set<String> orphanedLocations = MutableSet.of(
+                "banby1jx4j",
+                "msyp655po0",
+                "ppamsemxgo"
+        );
+
+        initManagementContextAndPersistence(persistenceDirWithOrphanedLocations);
+        BrooklynMementoRawData mementoRawData = managementContext.getRebindManager().retrieveMementoRawData();
+
+        Set<String> allReferencedLocationsFoundByTransformer = transformer.findAllReferencedLocations(mementoRawData);
+        Map<String, String> locationsToKeepFoundByTransformer = transformer.findLocationsToKeep(mementoRawData);
+
+        Asserts.assertEquals(allReferencedLocationsFoundByTransformer, locationsToKeep);
+        Asserts.assertEquals(locationsToKeepFoundByTransformer.keySet(), locationsToKeep);
+
+        Map<String, String> locationsNotToKeepDueToTransormer = MutableMap.of();
+        locationsNotToKeepDueToTransormer.putAll(mementoRawData.getLocations());
+        for (Map.Entry location: locationsToKeepFoundByTransformer.entrySet()) {
+            locationsNotToKeepDueToTransormer.remove(location.getKey());
+        }
+        Set<String> notReferencedLocationsDueToTransormer = MutableSet.of();
+        notReferencedLocationsDueToTransormer.addAll(mementoRawData.getLocations().keySet());
+        for (String location: allReferencedLocationsFoundByTransformer) {
+            notReferencedLocationsDueToTransormer.remove(location);
+        }
+
+        Asserts.assertEquals(notReferencedLocationsDueToTransormer, orphanedLocations);
+        Asserts.assertEquals(locationsNotToKeepDueToTransormer.keySet(), orphanedLocations);
+
+
+        BrooklynMementoRawData transformedMemento = transformer.transform(mementoRawData);
+        Asserts.assertFalse(EqualsBuilder.reflectionEquals(mementoRawData, transformedMemento));
```

Comment:
@iyovcheva you can use `CleanOrphanedLocationsTest#assertTransformDeletes` and `CleanOrphanedLocationsTest#assertTransformIsNoop`
I'll try to PR against your PR with so we can minify the size of methods in `CleanOrphanedLocationsLiveTest`